pkg/api/types/identity_provider.go

```diff
@@ -18,9 +25,71 @@ type IdentityProvider struct {
 	// TODO: add unique constraint for email domains
 	// TODO: add support for multiple email domains for a single IdentityProvider
 	EmailDomain string `json:"emailDomain"`
+
+	Scopes string `json:"scopes"`
+
+	Claim Claim `json:"claim"`
+}
+
+type Claim struct {
+	Version string `json:"Version"`
+	Mappings map[string]string `json:"Mappings"`
 }
 
 // IdentityProviderList represents a list of IdentityProvider
 type IdentityProviderList struct {
 	Items []*IdentityProvider `json:"items"`
 }
+
+func (c Claim) Value() (driver.Value, error) {
+	j, err := json.Marshal(c)
+	return j, err
+}
+
+func (c *Claim) Scan(src interface{}) error {
+	source, ok := src.([]byte)
+	if !ok {
+		return errors.New("type assertion .([]byte) failed")
```

Comment:
I would add a more helpful error message such as 
```
return fmt.Errorf("failed to scan scan claims: value of type %T could not be converted to []byte", src) 
```
that would also be helpful for the type assertions below